backend/src/test/java/com/example/backend/service/PetServiceTest.java

```diff
@@ -113,6 +113,30 @@ void when_addPet_and_add_Empty_Supplies_then_OK() {
 
     }
 
+    @Test
+    void when_updatePet_then_OK() {
+        //GIVEN
+        when(petRepo.save(pet1)).thenReturn(pet1);
+        when(petRepo.findById(pet1.id())).thenReturn(Optional.ofNullable(pet1));
+        //WHEN
+        Pet actual = petService.updatePet(pet1.id(), pet1DTO);
+        Pet expected = pet1;
+        //THEN
+        verify(petRepo).save(pet1);
+        verify(petRepo).findById(pet1.id());
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void when_updatePet_and_IdDoesntExist_then_throwException() {
+        //GIVEN
+        when(petRepo.existsById(pet1.id())).thenReturn(false);
+        when(petRepo.findById(pet1.id())).thenReturn(Optional.empty());
+        //WHEN & THEN
+        assertThrows(PetNotFoundException.class, () -> petService.updatePet(pet1.id(), pet1DTO));
```

Comment:
In order to avoid the code smell you can add the following line in your `WHEN` block: `String petId = pet1.id();`. Then you can call `updatePet(petId, pet1DTO)` and will not get the code smell :)